proto/types.proto

```diff
@@ -12,14 +12,59 @@ message ExtraData {
 	google.protobuf.Any RuncOptions = 2;
 }
 
+// Message to specify the various Firecracker VM configurations for a task
+message FirecrackerConfig {
+	google.protobuf.Any RuncOptions = 1; // Specifies Runc options that need to relayed to the CreateTask request
+	repeated FirecrackerNetworkInterface NetworkInterfaces = 2; // Specifies the networking configuration for a VM
+	string KernelImagePath = 3; // Specifies the file path where the kernel image is located
+	string KernelArgs = 4; // Specifies the commandline arguments that should be passed to the kernel
+	FirecrackerMachineConfiguration MachineCfg = 5; // Specifies the machine configuration for the VM
+	FirecrackerDrive RootDrive = 6; // Specifies the root block device for the VM
+	// Specifies the additional block device config for the VM. Note that
+	// the snapshotter used might itself present block devices to the be
+	// attached to the VM. This fields allows additional block devices to
+	// that list.
+	repeated FirecrackerDrive AdditionalDrives = 7;
+}
+
 // Message to specify network config for a Firecracker VM
 message FirecrackerNetworkInterface {
-	string MacAddress = 1;
-	string HostDevName = 2;
+	string MacAddress = 1; // Specifies the mac address for the the device
+	string HostDevName = 2; // Specifies the name of the tap device on the host
+	bool AllowMMDS = 3; // Specifies if metadata service should be available on this network interface
+	FirecrackerRateLimiter InRateLimiter = 4; // Specifies a rate limiter for incoming bytes
+	FirecrackerRateLimiter OutRateLimiter = 5; // Specifies a rate limiter for outgoing bytes
 }
 
-// Message to specify the various Firecracker VM configurations for a task
-message FirecrackerConfig {
-	google.protobuf.Any RuncOptions = 1; // Can be used by the CreateTask request needs to relay any Runc options to the agent
-	repeated FirecrackerNetworkInterface NetworkInterfaces = 2; // Can be used to set the networking configuration for a VM
+// Message to set the machine config for a Firecracker VM
+message FirecrackerMachineConfiguration {
+	string CPUTemplate = 1; // Specifies the cpu template. Example: "T2" or "C3"
+	bool HtEnabled = 2; // Specifies if hyper-threading should be enabled
+	 // Specifies the memory size of VM
+	 // This lets us create a Firecracker VM of up to 4096 TiB, which
+	 // for a mircroVM should be large enough
+	uint32 MemSizeMib = 3;
+	uint32 VcpuCount = 4; // Specifies the number of vCPUs for the VM
+}
+
+// Message to specify the block device config for a Firecracker VM
+message FirecrackerDrive {
+	bool IsReadOnly = 1; // Specifies if the drive is read only
+	bool IsRootDevice = 2; // Specifies if the drive is the root device
+	string Partuuid = 3; // Specifies the unique id of the boot partition on this device
```

Comment:
`PartUUID`?

Reply:
It's reflecting the name in https://github.com/firecracker-microvm/firecracker-go-sdk/blob/4b31e1c2d796c8a866fd5047ee689dfb03754a5f/client/models/drive.go#L46. I'm in favor of keeping it as is for the sake of consistency.